app/src/main/java/com/github/geohunt/app/ui/components/Challenge.kt

```diff
@@ -46,7 +49,7 @@ import com.github.geohunt.app.utility.findActivity
  * @param challenge The challenge to display
  */
 @Composable
-fun Challenge(challenge: Challenge, fnClaimCallback: (String) -> Unit = {}) {
+fun Challenge(challenge: Challenge, controller: NavController) {
```

Comment:
I think the `NavController` should not be passed as an argument to `Composable`, as it bloats the argument list (in Scala, you would probably declare a `given` instance and declare the controller as a `using` clause). The proper workaround in Kotlin would be to create a singleton instance, that you could retrieve in composable that need them via e.g. `NavController.getInstance()`.

Reply:
No, the controller should not be passed as argument. I forgot but I have seen somewhere in documentation that it should not be "shared".

Prefer callbacks.